src/components/Input/TagInputField.jsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import TopicTag from "../TopicTag/TopicTag.jsx";
+import "./Input.css";
+
+function TagInputField({
+  label,
+  placeholder = "Write something and press enter",
+  value = [],
+  onChange,
+  disabled = false,
+  invalid = false,
+}) {
+  const [inputValue, setInputValue] = useState("");
+
+  const handleKeyDown = (e) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      const trimmed = inputValue.trim();
+      if (trimmed !== "") {
+        const updated = [...value, trimmed];
+        onChange?.(updated);
+        setInputValue("");
+      }
+    }
+    if (
+      (e.key === "Backspace" || e.key === "Delete") &&
+      inputValue.trim() === "" &&
+      value.length > 0
+    ) {
+      e.preventDefault();
+      const updated = value.slice(0, -1);
+      onChange?.(updated);
+    }
+  };
+
+  const handleTagClick = (indexToRemove) => {
+    const updated = value.filter((_, i) => i !== indexToRemove);
+    onChange?.(updated);
+  };
+
+  return (
+    <div className={`input-field ${disabled ? "is-disabled" : ""}`}>
+      {label && <label className="input-label">{label}</label>}
+
+      <div
+        className={`input-wrapper tag-input-wrapper ${
+          invalid ? "is-invalid" : ""
+        }`}
+        onClick={() => {
+          const input = document.querySelector(".tag-input");
```

Comment:
if you have more than one element with `.tag-input` as class? if you're sure that there's only one, then you should use an id, i.e. "#tag-input".